packages/broker/src/plugins/legacyWebsocket/Connection.ts

```diff
@@ -1,97 +1,167 @@
 import { EventEmitter } from 'events'
 import { Logger, Protocol } from 'streamr-network'
-import uWS from 'uWebSockets.js'
 import { Stream } from '../../Stream'
+import WebSocket from "ws"
+import stream from "stream"
 
 const logger = new Logger(module)
 
 let nextId = 1
 
-function generateId() {
+function generateId(): string {
     const id = `socketId-${nextId}`
     nextId += 1
     return id
 }
 
-export class Connection extends EventEmitter {
-
-    static LOW_BACK_PRESSURE = 1024 * 1024 // 1 megabytes
-    static HIGH_BACK_PRESSURE = 1024 * 1024 * 2 // 2 megabytes
-
-    id: string
-    socket: uWS.WebSocket
-    streams: Stream[] = []
-    dead: boolean
-    controlLayerVersion: number
-    messageLayerVersion: number
-    highBackPressure: boolean
+export interface Connection {
+    on(eventName: 'message', handler: (msg: Protocol.ControlMessage) => void): this
+    on(eventName: 'close', handler: () => void): this
+    on(eventName: 'highBackPressure', handler: () => void): this
+    on(eventName: 'lowBackPressure', handler: () => void): this
+}
 
-    constructor(socket: uWS.WebSocket, controlLayerVersion: number, messageLayerVersion: number) {
+export class Connection extends EventEmitter {
+    readonly id: string
+    readonly controlLayerVersion: number
+    readonly messageLayerVersion: number
+    private readonly socket: WebSocket
+    private readonly duplexStream: stream.Duplex
+    private readonly streams: Stream[] = []
+    private dead = false
+    private highBackPressure = false
```

Comment:
The value of `highBackPressure` field is never read

Reply:
I was gonna remove it but I started thinking it may be useful to have such a field for debugging purposes? OTOH it's not used anywhere, dunno what do you think?

Reply:
Maybe easier to remove. Add logging if needed?